core/kogitoservice/truststore.go

```diff
@@ -0,0 +1,124 @@
+	trustStoreSecretFileKey      = "cacerts"
+	trustStoreEnvVarPassword     = "CUSTOM_TRUSTSTORE_PASSWORD"
+	trustStoreEnvVarCertFileName = "CUSTOM_TRUSTSTORE"
+	trustStoreVolumeName         = "custom-truststore"
+)
+
+// TrustStoreHandler takes care of mounting the custom TrustStoreSecret in a given Deployment based on api.KogitoService spec
+type TrustStoreHandler interface {
+	MountTrustStore(deployment *appsv1.Deployment, service api.KogitoService) error
+}
+
+// NewTrustStoreHandler creates a new TrustStoreHandler with the given context
+func NewTrustStoreHandler(context *operator.Context) TrustStoreHandler {
+	return &trustStoreHandler{
+		context: context,
+	}
+}
+
+type trustStoreHandler struct {
+	context *operator.Context
+}
+
+// MountTrustStore mounts the given custom TrustStoreSecret based on api.KogitoService
+func (t *trustStoreHandler) MountTrustStore(deployment *appsv1.Deployment, service api.KogitoService) error {
+	if len(service.GetSpec().GetTrustStoreSecret()) == 0 {
+		return nil
+	}
+
+	secret, err := t.fetchAndValidateTrustStoreSecret(service)
+	if err != nil {
+		return err
+	}
+
+	t.mapTrustStorePassword(deployment, secret)
+	if err := t.mountTrustStoreFile(deployment, secret); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *trustStoreHandler) fetchAndValidateTrustStoreSecret(service api.KogitoService) (*v1.Secret, error) {
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      service.GetSpec().GetTrustStoreSecret(),
+			Namespace: service.GetNamespace(),
+		},
+	}
+	if exists, err := kubernetes.ResourceC(t.context.Client).Fetch(secret); err != nil {
+		return nil, err
+	} else if !exists {
+		return nil, errorForTrustStoreMount("Failed to find Secret named " + secret.Name + " in the namespace " + secret.Namespace)
+	}
+
+	if _, ok := secret.Data[trustStoreSecretFileKey]; !ok {
+		return nil, errorForTrustStoreMount("Failed to mount Truststore. Secret " + secret.Name + " doesn't have the file with key " + trustStoreSecretFileKey)
+	}
+
+	return secret, nil
+}
+
+func (t *trustStoreHandler) mountTrustStoreFile(deployment *appsv1.Deployment, secret *v1.Secret) error {
+	trustStoreVolume := v1.Volume{
+		Name: trustStoreVolumeName,
+		VolumeSource: v1.VolumeSource{
+			Secret: &v1.SecretVolumeSource{
+				SecretName:  secret.Name,
+				Items:       []v1.KeyToPath{{Key: trustStoreSecretFileKey, Path: trustStoreSecretFileKey}},
+				DefaultMode: &framework.ModeForCertificates,
+			},
+		},
+	}
+	trustStoreMount := v1.VolumeMount{
+		Name:      trustStoreVolumeName,
+		MountPath: trustStoreMountPath + "/" + trustStoreSecretFileKey,
+		SubPath:   trustStoreSecretFileKey,
+	}
+
+	framework.AddVolumeToDeployment(deployment, trustStoreMount, trustStoreVolume)
+	deployment.Spec.Template.Spec.Containers[0].Env = framework.EnvOverride(deployment.Spec.Template.Spec.Containers[0].Env, v1.EnvVar{
+		Name:  trustStoreEnvVarCertFileName,
+		Value: trustStoreSecretFileKey,
```

Comment:
Is it mandatory that user always create its custom trust store with name `cacerts`?

Reply:
Yes

Reply:
Then why we are creating its env var. If it fixed then we could hardcode it in images.

Reply:
In the images are not hardcoded :/

But you're right, could be anything. I avoided that here to make things simpler since this secret could have other information required by users in a migration scenario, for instance.